src/supernova/mod.rs

```diff
@@ -593,6 +614,31 @@ where
     let r_U_primary_initial_list = (0..num_augmented_circuits)
       .map(|i| (i == circuit_index).then(|| r_U_primary.clone()))
       .collect::<Vec<Option<RelaxedR1CSInstance<E1>>>>();
+
+    // find the largest length r1cs shape for the buffer size
+    let mut max_r1cs_primary = &pp[0].r1cs_shape;
+    pp.circuit_shapes.iter().for_each(|circuit| {
+      if circuit.r1cs_shape.num_cons > max_r1cs_primary.num_cons {
+        max_r1cs_primary = &circuit.r1cs_shape;
+      }
+    });
+
+    let buffer_primary = ResourceBuffer {
+      l_w: None,
+      l_u: None,
+      ABC_Z_1: R1CSResult::default(max_r1cs_primary),
+      ABC_Z_2: R1CSResult::default(max_r1cs_primary),
+      T: r1cs::default_T(max_r1cs_primary),
+    };
```

Comment:
`R1CSResult::default()` only requires the `num_cons` value from the `R1CSShape`. If we change the function to take this value instead, then we can more simply compute

```rust
let max_num_cons = *pp.circuit_shapes.iter().map(|shape| shape.num_cons).max().unwrap();
let buffer_primary = ResourceBuffer {
      l_w: None,
      l_u: None,
      ABC_Z_1: R1CSResult::default(max_num_cons),
      ABC_Z_2: R1CSResult::default(max_num_cons),
      T: r1cs::default_T(max_num_cons),
    }; 
```